src/odemis/acq/align/shift.py

```diff
@@ -19,167 +19,158 @@
-                                              )
-
-        # Cross-correlation computation
-        CC = fft.ifft2(fft.ifftshift(CC))
-
-        # Locate the peak
-        ACC = abs(CC)
-        loc1 = ACC.argmax(0)
-        max1 = ACC[(loc1, range(ACC.shape[1]))]
-        loc2 = max1.argmax(0)
-
-        rloc = loc1[loc2]
-        cloc = loc2
-
-        # Calculate shift in previous pixel grid from the position of the peak
-        (m, n) = CC.shape
-        md2 = m // 2
-        nd2 = n // 2
-
-        if rloc > md2:
-            row_shift = rloc - m
-        else:
-            row_shift = rloc
-
-        if cloc > nd2:
-            col_shift = cloc - n
-        else:
-            col_shift = cloc
-
-        row_shift /= 2
-        col_shift /= 2
-
-        # DFT computation
-        # Initial shift estimation in upsampled grid
-        row_shift = round(row_shift * precision) / precision
-        col_shift = round(col_shift * precision) / precision
-        dft_shift = math.ceil(precision * 1.5) // 2  # Center of output at dft_shift+1
-
-        # Matrix multiply DFT around the current shift estimation
-        CC = (_UpsampledDFT(current_fft * previous_fft.conj(),
-                            math.ceil(precision * 1.5),
-                            math.ceil(precision * 1.5),
-                            precision,
-                            dft_shift - row_shift * precision,
-                            dft_shift - col_shift * precision)
-              ) / (md2 * nd2 * (precision ** 2))
-        # was .conj(), but as we just need the abs(), it's not needed
-
-        # Locate maximum and map back to original pixel grid
-        ACC = abs(CC)
-        loc1 = ACC.argmax(0)
-        max1 = ACC[(loc1, range(ACC.shape[1]))]
-        loc2 = max1.argmax(0)
-
-        rloc = loc1[loc2]
-        cloc = loc2
+        row_shift = rloc
 
-        rloc -= dft_shift
-        cloc -= dft_shift
-
-        row_shift += rloc / precision
-        col_shift += cloc / precision
-
-        if md2 == 1:
-            row_shift = 0
-        if nd2 == 1:
-            col_shift = 0
+    if cloc > nd2:
+        col_shift = cloc - n
+    else:
+        col_shift = cloc
 
-    return col_shift, row_shift
+    shifts = [row_shift, col_shift]
 
+    if precision == 1:
+        pass
 
-def _UpsampledDFT(data, nor, noc, precision=1, roff=0, coff=0):
-    """
-    Upsampled DFT by matrix multiplies.
-    data (numpy.array): 2d array
-    nor, noc (ints): Number of pixels in the output upsampled DFT, in units
-    of upsampled pixels
-    precision (int): Calculate drift within 1/precision of a pixel
-    roff, coff (ints): Row and column offsets, allow to shift the output array
-                    to a region of interest on the DFT
-    returns (tuple of floats): Drift in pixels
-    """
-    z = 1j  # imaginary unit
-    nr, nc = data.shape
+    else:
+        precision_factor = numpy.array(precision, dtype=float_dtype)
```

Comment:
Can you add a short comment explaining the trick of this line? It seems to just take the Python float and convert it to a numpy float of the "right" type. What's the advantage? Wouldn't numpy do the right thing otherwise?(typically numpy is fine with doing numpy.array * python_float)

Reply:
I have removed this line as Python does convert the result between float and int to float. I also do not think this line is necessary. Probably in other languages like C this is an important step. I have tested without it  and it works similar.